src/js/tech/tech.js

```diff
@@ -449,6 +449,42 @@ class Tech extends Component {
            component instanceof Tech ||
            component === Tech;
   }
+
+  /**
+   * Registers a Tech
+   *
+   * @param {String} name Name of the Tech to register
+   * @param {Object} tech The tecj to register
+   * @static
+   * @method registerComponent
+   */
+  static registerTech(name, tech) {
+    if (!Tech.techs_) {
+      Tech.techs_ = {};
+    }
+
+    Tech.techs_[name] = tech;
```

Comment:
Now that I'm thinking about it, should we do that?
Is a Tech specifically something that inherits from `Tech` or can it be anything that follows the same API?